presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -0,0 +1,252 @@
+        }
+    }
+
+    private List<HostAddress> getSplitAddresses(PhoenixInputSplit split)
+    {
+        List<HostAddress> addresses;
+        try {
+            String hostName = split.getLocations()[0];
+            HostAddress address = hostCache.get(hostName);
+            if (address == null) {
+                address = HostAddress.fromString(hostName);
+                hostCache.put(hostName, address);
+            }
+            addresses = ImmutableList.of(address);
+        }
+        catch (Exception e) {
+            log.warn("Failed to get split host addresses, will proceed without locality");
+            addresses = ImmutableList.of();
+        }
+        return addresses;
+    }
+
+    // use Phoenix MR framework to get splits
+    private List<InputSplit> getHadoopInputSplits(String inputQuery)
+            throws SQLException, IOException, InterruptedException
+    {
+        Configuration conf = new Configuration();
+        phoenixClient.setJobQueryConfig(inputQuery, conf);
+        Job job = Job.getInstance(conf);
+        return getSplits(job);
+    }
+
+    public List<InputSplit> getSplits(JobContext context)
+            throws IOException, InterruptedException
+    {
+        final Configuration configuration = context.getConfiguration();
+        final QueryPlan queryPlan = getQueryPlan(context, configuration);
+        final List<KeyRange> allSplits = queryPlan.getSplits();
+        return generateSplits(queryPlan, allSplits, configuration);
+    }
+
+    // mostly copied from PhoenixInputFormat, but without the region size calculations
+    private List<InputSplit> generateSplits(final QueryPlan qplan, final List<KeyRange> splits, Configuration config)
+            throws IOException
+    {
+        Preconditions.checkNotNull(qplan);
+        Preconditions.checkNotNull(splits);
+
+        try (org.apache.hadoop.hbase.client.Connection connection =
+                HBaseFactoryProvider.getHConnectionFactory().createConnection(config)) {
+            RegionLocator regionLocator = connection.getRegionLocator(TableName.valueOf(qplan.getTableRef().getTable().getPhysicalName().toString()));
+            long regionSize = -1;
+            final List<InputSplit> psplits = Lists.newArrayListWithExpectedSize(splits.size());
+            for (List<Scan> scans : qplan.getScans()) {
+                // Get the region location
+                HRegionLocation location = regionLocator.getRegionLocation(scans.get(0).getStartRow(), false);
+                String regionLocation = location.getHostname();
+
+                // Generate splits based off statistics, or just region splits?
+                boolean splitByStats = PhoenixConfigurationUtil.getSplitByStats(config);
+
+                if (splitByStats) {
+                    for (Scan aScan : scans) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Split for  scan : " + aScan + "with scanAttribute : " +
```

Comment:
Use `%s` placeholders.
Then, is `if (log.isDebugEnabled())` required?

Reply:
To expand on the above comment, the Airlift logger uses standard Java string formatting. An easy way to fix these in IntelliJ is to put the cursor on the `+`, hit alt-enter for quick fix, choose *"Replace with String.format()"*, then remove the outer `format()` call.

The string format is only evaluated if the log level is enabled. Thus, assuming all the calls like `aScan.getBatch()` are cheap, there's no need to conditionally call the log method. Of course, this would not be true for things like `Bytes.toStringBinary()`, since that would be evaluated every time.